README.md

```diff
@@ -262,30 +262,126 @@ Parser consists of the following phases:
 
-### Parsing HTML
+### Data flow
+An extension's architecture includes such components as:
+
+background script that contains listeners for browser events and communication with Redux
+content script that contains JavaScript that executes in the context of a page that has been loaded in the browser.
+Depending on the format (JSON or HTML markup), fragments' debug data is processed by the content script first and then stored in Redux with the background script (which basically wraps Redux store). Content script transformations are described [here](https://github.com/Knotx/knotx-fragments-chrome-extension#debug-fragment-data).
+
+```
+KNOT.x -> HTML MARKUP -> CONTENT SCRIPT -> BACKGROUND SCRIPT -> REDUX -> COMPONENTS
+                               ^
+                               |
+                         Parsing helper
+```
 
 ### Components
+The components structure in the main concept look like this:
+
+```
+•
+└── App:
+    ├── SidePanel
+    │   ├── FragmenList
+    │   │   └── FragmentListItem
+    │   │       └── NodeList
+    │   └── FragmentGannt
+    │
+    └── MainPanel
+        └── Graph
+            ├──  Timeline
+            ├──  Legend
+            │    └── LegendSection
+            └──  NodeInfo
+```
+
+#### Graph && timelines
+To create a data visualization we use vis.js library. Currently we have 3 charts in our application.
+
+* Timeline for fragments on sidePanel (FragmentGannt component)
+* Main graph for the fragment (Graph component)
+* Performance timeline for the fragment (Timeline component)
 
-#### Graph
+### Styling
+We don't use any grid system to make our app beautiful. Everything is flex. To show and hide elements we try to use a react state, without saving this information in the redux store.
+
+To create styles we use styling-component .We follow the convention to create a style file next to js file.
+
+ ```
+•
+├── exampleComponent.js
+└── exampleComponent.style.js
+```
+
+some global styling and styling for renderjson markupwe store in
+ ```
+/src/js/styling/globalStyle.js
+  ```
 
-#### Timelines
 
 ### Data storage
+All data about fragments and pages are store in redux. We store a separate data set and an app state for all active pages to allow using a few dev tools console for a few Knot.x pages. Every pageData-set has own ID (chrome tab id). The structure of redux looks like this:
 
-## Contributors
 
-### How to run?
-1. go to the project folder
-2. run command: `yarn install`
-3. run command: `yarn run dev`
+```
+•
+└── pageData:
+    ├── 78: // tab id
+    │   ├── fragments: [] // list of fragments
+    │   ├── url: "https://example.com // page url
+    │   ├── sidebarExpanded: true // side panel expanded switch
+    │   └── renderedGraph: null // id of the currently selected fragment
+    └── 110:
+        └── ...
+```
+
+The pageData item is created on page load and destroy on page close. We create an item for all active pages. If the page doesn't use Knot.x , fragments property is empty.
```

Comment:
The pageData entry is created on page load and destroyed when we close the tab. If the page does not contain Knot.x fragments, `fragments` property is empty.